python/ray/autoscaler/_private/autoscaler.py

```diff
@@ -127,13 +128,17 @@ def __init__(self, provider: NodeProvider):
         self.worker_ids: List[NodeID] = []
         # The head node (node kind "head")
         self.head_id: Optional[NodeID] = None
+        # Map of replica IDs to worker nodes in each replica
+        self.replicas_to_nodes = defaultdict(list)
 
         for node in self.all_node_ids:
             node_kind = provider.node_tags(node)[TAG_RAY_NODE_KIND]
             if node_kind == NODE_KIND_WORKER:
                 self.worker_ids.append(node)
             elif node_kind == NODE_KIND_HEAD:
                 self.head_id = node
+            node_replica_index = provider.node_tags(node)[TAG_RAY_REPLICA_INDEX]
```

Comment:
What if there are multiple worker groups? Would the single index cause any confusion here?

Reply:
`TAG_RAY_REPLICA_INDEX` is set to the `replicaIndex` label value, which is of the form `{$WORKER_GROUP_NAME}-{$INDEX}`, this should handle any confusion with multiple worker groups.